src/views/HomeView.vue

```diff
@@ -6,16 +6,24 @@ import { onBeforeMount } from "vue";
 import { useRoute } from "vue-router";
 import { useSearchStore } from "../stores/searchStore.js";
 import { watchEffect } from 'vue';
+import {useTagsStore} from "../stores/tagsStore.js";
 
 const route = useRoute();
 const { t } = i18n.global;
+const tagsStore = useTagsStore();
 const searchStore = useSearchStore();
 const { search } = route.query;
 
 onBeforeMount(() => {
 	if (search?.length) {
 		searchStore.setSearchText(search);
 	}
+
+	if (route.query.tags?.length) {
+		const tags = route.query.tags.split(',');
+
+		tags.forEach(tag => tagsStore.selectTag(tag));
```

Comment:
correct me if I'm wrong. I did not check the behavior of this PR manually. On the other side I suppose here is a bug.

Steps to reproduce: 
1. Create a Tags filter with the values `programming`, `assistant`.
- As result. localStorage is storing the values `programming`, `assistant`.
Url should be like `?tags=programming,assistant`. Let's save it in some notepad.

2. Change a new Tags filter with values `google`, `llm`.
- As result. localStorage is storing the values `google`, `llm`.
Url should be like `?tags= google,llm`.

3. Go to the URL from the first step (`?tags=programming,assistant` )
- As result. localStorage is storing the values `google`, `llm`, `programming`, `assistant`.
Url should be... not sure about the correct value. And actually it does not matter.

Actual tags filter state has 4 values instead of 2 values (correct value).

Reply:
It happens due to `vue`/`pinia` store lifecycles. 
Before any store-usage Pinia is creating it and initialize its `state`. But state is initialized by `localStorage` values.
On the other side I would like to avoid to modify any user filters by such urls. Seems it is a bad UX.